### PR TITLE
[1.19.2] Added PauseGameEvent

### PR DESCRIPTION
**Pull request by SrRapero720**
same as #9761

### CODE REVIEW

**Comment by SrRapero720** [comment]
build checks just die.

File: fmlloader/src/main/java/net/minecraftforge/fml/loading/ModSorter.java [hunk 1]
```diff
@@ -256,4 +256,4 @@ private boolean modVersionNotContained(final IModInfo.ModVersion mv, final Map<S
         return !(VersionSupportMatrix.testVersionSupportMatrix(mv.getVersionRange(), mv.getModId(), "mod", (modId, range) -> modVersions.containsKey(modId) &&
                 (range.containsVersion(modVersions.get(modId)) || modVersions.get(modId).toString().equals("0.0NONE"))));
     }
-}
+}
```
Comment (Jonathing) [inline fmlloader/src/main/java/net/minecraftforge/fml/loading/ModSorter.java]: Unnecessary newline deletion.

File: src/main/java/net/minecraftforge/event/ForgeEventFactory.java [hunk 2]
```diff
@@ -868,4 +874,4 @@ public static void onAdvancementProgressedEvent(Player player, Advancement progr
     {
         MinecraftForge.EVENT_BUS.post(new AdvancementProgressEvent(player, progressed, advancementProgress, criterion, progressType));
     }
-}
+}
```
Comment (Jonathing) [inline src/main/java/net/minecraftforge/event/ForgeEventFactory.java:818]: Unnecessary newline deletion.

File: src/main/java/net/minecraftforge/event/ForgeEventFactory.java [hunk 3]
```diff
@@ -799,6 +800,11 @@ public static void firePlayerSmeltedEvent(Player player, ItemStack smelted)
         MinecraftForge.EVENT_BUS.post(new PlayerEvent.ItemSmeltedEvent(player, smelted));
     }
 
+    public static void firePauseGameEvent(boolean paused)
+    {
+        MinecraftForge.EVENT_BUS.post(new ClientPauseEvent(paused));
+    }
+
```
Comment (Jonathing) [inline src/main/java/net/minecraftforge/event/ForgeEventFactory.java]: Please move this event to `net.minecraftforge.client.ForgeHooksClient`, as this class only contains common event factories and firing.

**Review by Jonathing — CHANGES_REQUESTED** [review]
Please see my comments on the main PR: https://github.com/MinecraftForge/MinecraftForge/pull/9763#pullrequestreview-1714011458

File: src/main/java/net/minecraftforge/event/ForgeEventFactory.java (hunk 3 above)
Comment (SrRapero720) [reply src/main/java/net/minecraftforge/event/ForgeEventFactory.java]: done

File: src/main/java/net/minecraftforge/event/ForgeEventFactory.java (hunk 2 above)
Comment (SrRapero720) [reply src/main/java/net/minecraftforge/event/ForgeEventFactory.java:818]: https://github.com/MinecraftForge/MinecraftForge/pull/9763#discussion_r1382621647

**Comment by SrRapero720** [comment]
moved to #9781

**Pull request closed without merging**